util/types/field_type.go

```diff
@@ -285,7 +288,7 @@ func DefaultTypeForValue(value interface{}, tp *FieldType) {
 	case bool:
 		tp.Tp = mysql.TypeLonglong
 		tp.Flen = 1
-		tp.Decimal = 0
+		tp.Decimal = IntIsBoolean
```

Comment:
I think `Decimal` field should not be assigned to a flag value `IntIsBoolean`.

We can add a flag to indicate this is a boolean, like this
https://github.com/pingcap/tidb/blob/master/mysql/type.go#L67